src/middlewares/patient.js

```diff
@@ -3,18 +3,26 @@
 
     var keyFor = {
         "Observation": "subject",
-        "MedicationPrescription": "patient"
+        "MedicationPrescription": "patient",
+        "Encounter": "patient"
```

Comment:
But we technically don't even need to any more -- at least, not strictly.
We can just tack a patient search param on whenever this middleware is
active. I was imagining that SMART's client at least would expose two
instances of fhir.js client:

 * smart.api  --> has patient middleware disabled
 * smart.patient.api --> has patient middleware enabled, for the in-context
patient.

Maybe this is over-thinking things, but it's convenient not to require an
explicit "patient" parameter all over the place.

then users can call whichever one they want in a given context.

On Thu, Sep 3, 2015 at 9:05 AM, niquola <notifications@github.com> wrote:

> In src/middlewares/patient.js
> <https://github.com/FHIR/fhir.js/pull/50#discussion_r38642730>:
>
> > @@ -3,18 +3,26 @@
> >
> >      var keyFor = {
> >          "Observation": "subject",
> > -        "MedicationPrescription": "patient"
> > +        "MedicationPrescription": "patient",
> > +        "Encounter": "patient"
>
> We can extract from FHIR metadata all refs to Patient
> … <#14f934fc554f0459_>
> On 3 September 2015 at 16:02, nschwertner <notifications@github.com>
> wrote: In src/middlewares/patient.js <
> https://github.com/FHIR/fhir.js/pull/50#discussion_r38642475>: > @@ -3,18
> +3,26 @@ > > var keyFor = { > "Observation": "subject", > -
> "MedicationPrescription": "patient" > + "MedicationPrescription":
> "patient", > + "Encounter": "patient" Is there a definitive list of the
> resources "with a patient" in DSTU2? — Reply to this email directly or view
> it on GitHub <https://github.com/FHIR/fhir.js/pull/50/files#r38642475>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/FHIR/fhir.js/pull/50/files#r38642730>.
>